src/components/ec-timer/ec-timer.spec.js

```diff
@@ -0,0 +1,65 @@
+import { mount } from '@vue/test-utils';
+import EcTimer from './ec-timer.vue';
+import { withMockedConsole } from '../../../tests/utils/console';
+
+jest.useFakeTimers();
+
+describe('EcTimer', () => {
+  function mountTimer(props) {
+    return mount(EcTimer, {
+      propsData: {
+        ...props,
+      },
+    });
+  }
+
+  it('should throw an error if "seconds" prop is not given', () => {
+    withMockedConsole((errorSpy) => {
+      mountTimer({ isTimerRunning: true });
+
+      expect(errorSpy).toHaveBeenCalledTimes(1);
+      expect(errorSpy.mock.calls[0][0]).toContain('Missing required prop: "seconds"');
+    });
+  });
+
+  it('should throw an error if "seconds" prop is not an integer', () => {
+    withMockedConsole((errorSpy) => {
+      mountTimer({ seconds: 20.1, isTimerRunning: true });
+
+      expect(errorSpy).toHaveBeenCalledTimes(1);
+      expect(errorSpy.mock.calls[0][0]).toContain('Invalid prop: custom validator check failed for prop "seconds"');
+    });
+  });
+
+  it('should throw an error if "seconds" prop is negative ', () => {
+    withMockedConsole((errorSpy) => {
+      mountTimer({ seconds: -20, isTimerRunning: true });
+
+      expect(errorSpy).toHaveBeenCalledTimes(1);
+      expect(errorSpy.mock.calls[0][0]).toContain('Invalid prop: custom validator check failed for prop "seconds"');
+    });
+  });
+
+  it('should throw an error if "isTimerRunning" prop is not given', () => {
+    withMockedConsole((errorSpy) => {
+      mountTimer({ seconds: 20 });
+
+      expect(errorSpy).toHaveBeenCalledTimes(1);
+      expect(errorSpy.mock.calls[0][0]).toContain('Missing required prop: "isTimerRunning"');
+    });
+  });
+
+  it('should render as expected', () => {
+    const wrapper = mountTimer({ seconds: 20, isTimerRunning: true });
+
+    expect(wrapper.element).toMatchSnapshot();
+  });
+
+  it('should emit an event called "time-expired" after the countdown completes', () => {
+    const wrapper = mountTimer({ seconds: 20, isTimerRunning: true });
+
+    jest.advanceTimersByTime(20000);
```

Comment:
Before I merge I need to figure out why I am getting the following error when I use advanceTimerByTime
![image](https://user-images.githubusercontent.com/55188371/99246142-b2cdb680-27fc-11eb-8107-fa618f80ee87.png)

Is like it doesn't have the time to validate?

Reply:
Maybe it is because invoking `jest.useFakeTimers();` at the root level makes that it applies for all tests, but actually it is needed only for the last one.
It seems that if you remove this line, and include it inside the test case, it works:
![image](https://user-images.githubusercontent.com/4802418/99251195-37283580-280d-11eb-8c5a-99b9cfc2298a.png)

Reply:
Thank you! Works pushiiiing it!